src/lakefs_spec/spec.py

```diff
@@ -88,42 +84,43 @@ def __init__(
         verify_ssl: bool = True,
         ssl_ca_cert: str | None = None,
         proxy: str | None = None,
-        configfile: str = "~/.lakectl.yaml",
         create_branch_ok: bool = True,
         source_branch: str = "main",
         **storage_options: Any,
     ):
         super().__init__(**storage_options)
 
-        if (p := Path(configfile).expanduser()).exists():
-            lakectl_config = LakectlConfig.read(p)
+        # lakeFS client arguments
+        cargs = [host, username, password, api_key, api_key_prefix, access_token, ssl_ca_cert]
+
+        if all(arg is None for arg in cargs):
+            # empty kwargs means envvar and configfile autodiscovery
+            self.client = Client()
         else:
-            # empty config.
-            lakectl_config = LakectlConfig()
-
-        configuration = Configuration(
-            host=host or os.getenv("LAKEFS_HOST") or lakectl_config.host,
-            api_key=api_key or os.getenv("LAKEFS_API_KEY"),
-            api_key_prefix=api_key_prefix or os.getenv("LAKEFS_API_KEY_PREFIX"),
-            access_token=access_token or os.getenv("LAKEFS_ACCESS_TOKEN"),
-            username=username or os.getenv("LAKEFS_USERNAME") or lakectl_config.username,
-            password=password or os.getenv("LAKEFS_PASSWORD") or lakectl_config.password,
-            ssl_ca_cert=ssl_ca_cert or os.getenv("LAKEFS_SSL_CA_CERT"),
-        )
+            # TODO: `lakefs` does not consider configfiles other than ~/.lakectl.yaml,
+            #  do we keep any file or do we drop configfile support and defer?
```

Comment:
I don't see where other configurations would be necessary. So I'd be in favour of deferment

Reply:
Removed `lakefs_spec.config` in [71bd77d](https://github.com/aai-institute/lakefs-spec/pull/231/commits/71bd77d2717372b1f4f9ff296aa7ab9dbeff9362).